crates/ra_ide_api/src/name_kind.rs

```diff
@@ -20,14 +20,14 @@ pub(crate) fn classify_name_ref(
     db: &RootDatabase,
     analyzer: &hir::SourceAnalyzer,
     name_ref: &ast::NameRef,
-) -> Option<NameRefKind> {
-    use NameRefKind::*;
+) -> Option<NameKind> {
```

Comment:
I wonder if we should just call this a `Declaration` and implement various presentation-like functions on this type? Definitely out of scope for this PR